tools/clang/test/CodeGenDXIL/reflection/type_sys_dump_mat_as_vector_2.ll

```diff
@@ -0,0 +1,114 @@
+; RUN: %dxa %s -o %t
+; RUN: %dxa %t -dumpreflection | FileCheck %s
+
+target datalayout = "e-m:e-p:32:32-i1:32-i8:32-i16:32-i32:32-i64:64-f16:32-f32:32-f64:64-n8:16:32:64"
+target triple = "dxil-ms-dx"
+
+;struct S2 {
+;   int4x3 m;
+;	row_major int1x3 m1[2];
+;	row_major int3x1 m2;
```

Comment:
Since row_major/column_major makes no difference to the type at the moment, shouldn't we verify that it works whether or not the llvm type is expressed in a transposed way?  You were doing that (accidentally) before.  I'd hate to increase the number of variants of this test case, since it's already getting unwieldy.  The _2 variants appear to just swap row_major and column_major (default) on fields in HLSL, but the llvm form is identical  That's good to test once, but could be tested separately from other dimension tests, right?  In other words, we could have one test that ensures type annotation has correct layout for row or column major, and that reflection is handled whether that layout is expressed in llvm as the row or column major.  Then we shouldn't need each test to have the `_2` variant that tests with the HLSL row/column major swapped, but same IR otherwise for each test scenario.